pkg/api/networkservice/mechanisms/kernel/helpers.go

```diff
@@ -75,3 +77,26 @@ func (m *mechanism) GetPCIAddress() string {
 func (m *mechanism) IsPCIDevice() bool {
 	return m.GetPCIAddress() == ""
 }
+
+// GetInterfaceName - returns the Kernel Interface Name
+//                    this is Mechanism.Parameters[InterfaceNameKey] if set
+//                    otherwise returns a name computed from networkservice.Connection 'conn'
+func (m *mechanism) GetInterfaceName(conn *networkservice.Connection) string {
+	if m == nil || m.GetParameters()[InterfaceNameKey] == "" {
+		ns := conn.GetNetworkService()
+		nsMaxLength := LinuxIfMaxLength - 5
+		if len(ns) > nsMaxLength {
+			ns = ns[:nsMaxLength]
+		}
+		name := fmt.Sprintf("%s-%s", ns, conn.GetId())
```

Comment:
is there any possibility this will cause problems if someone requests an interface that results in too long of a name and we end up with multiple interfaces that get the conn id truncated w/name collisions?

Reply:
It's not impossible, but unlikely.  And if it does occur it will cause an error in the Forwarder which will result in a retry where you are lucky to get luckier :)